Library/Homebrew/os.rb

```diff
@@ -54,7 +54,8 @@ def self.kernel_name
     if !OS::Mac.version.prerelease? &&
        !OS::Mac.version.outdated_release? &&
        ARGV.none? { |v| v.start_with?("--cc=") } &&
-       ENV["HOMEBREW_PREFIX"] == "/usr/local"
+       (ENV["HOMEBREW_PREFIX"] == HOMEBREW_DEFAULT_PREFIX ||
+       (ENV["HOMEBREW_PREFIX"] == HOMEBREW_MACOS_ARM_DEFAULT_PREFIX && Hardware::CPU.arm?))
```

Comment:
```suggestion
       ENV["HOMEBREW_PREFIX"] == Homebrew::DEFAULT_PREFIX
```
I think this might work.

Reply:
`default_prefix.rb` is loaded after `os.rb`, because it uses OS conditionals.